src/ui/router/ApplicationRouterSpec.js

```diff
@@ -28,6 +28,9 @@ describe('Application router utility functions', () => {
             };
 
             openmct.router.on('change:hash', resolveFunction);
+            // We have a debounce set to 300ms on setHash, so if we don't flush,
+            // the above resolve function sometimes doesn't fire due to a race condition.
+            openmct.router.setHash.flush();
```

Comment:
wow, this is great. I'll add the flaky test closeout to this PR

Reply:
Hopefully it flakes less! I hate race conditions